tools/serverpod_cli/lib/src/analyzer/models/validation/model_validator.dart

```diff
@@ -42,6 +43,29 @@ List<SourceSpanSeverityException> validateTopLevelModelType(
   return errors;
 }
 
+List<SourceSpanSeverityException> validateDuplicateFileName(
+  YamlMap documentContents,
+  Restrictions restrictions,
+) {
+  var model = restrictions.documentDefinition;
+  var parsedModels = restrictions.parsedModels;
+
+  if (model == null) return [];
+
+  if (!parsedModels.isFilePathUnique(model)) {
+    var otherClass = parsedModels.findByFilePath(model, ignore: model);
+
+    return [
+      SourceSpanSeverityException(
+        'File collision with "${otherClass.className}" was detected for the generated model, please provide a unique path or filename for the model.',
```

Comment:
Suggestion: This could be made a little more clear for the end user.

"File path", "File name" collision for "classname, with path: $path" collides with "classname, with path $path"

Something like that.

Reply:
To make this really clear for the user we should include the path to the colliding model definition. In order to to that we need to refactor how we store models in the stateful analyzer. I'll give this a stab in a separate PR and improve the message once that is done.